lib/compiler/src/beam_types.erl

```diff
@@ -1011,6 +1046,67 @@ gcd(A, B) ->
         X -> gcd(B, X)
     end.
 
+%%%
+%%% Handling of ranges.
+%%%
+%%% A range can begin with '-inf' OR end with '+inf'.
+%%%
+%%% Atoms are greater than all integers. Therefore, we don't
+%%% need any special handling of '+inf'.
+%%%
+
+float_from_range(none) ->
+    none;
+float_from_range(any) ->
+    #t_float{};
+float_from_range({'-inf','+inf'}) ->
+    #t_float{};
+float_from_range({'-inf',Max}) ->
+    #t_float{elements={'-inf',float(Max)}};
+float_from_range({Min,'+inf'}) ->
+    #t_float{elements={float(Min),'+inf'}};
+float_from_range({Min,Max}) ->
+    #t_float{elements={float(Min),float(Max)}}.
+
+integer_from_range(none) ->
+    none;
+integer_from_range(any) ->
+    #t_integer{};
+integer_from_range({'-inf','+inf'}) ->
+    #t_integer{};
+integer_from_range({'-inf',Max}) ->
+    #t_integer{elements={'-inf',ceil(Max)}};
+integer_from_range({Min,'+inf'}) ->
+    #t_integer{elements={floor(Min),'+inf'}};
+integer_from_range({Min,Max}) ->
+    #t_integer{elements={floor(Min),ceil(Max)}}.
+
+number_from_range(none) ->
+    none;
```

Comment:
Isn't this handled by `integer_from_range/1` too?

Reply:
Yes, I intended to remove it but forgot.